fboom_me.sh

```diff
@@ -0,0 +1,201 @@
+        return $ERR_LINK_DEAD
+    fi
+
+    # File only available to premium members
+    if match 'only for premium members.' "$PAGE"; then
+        return $ERR_LINK_NEED_PERMISSIONS
+    fi
+
+    FILE_NAME=$(parse 'Download file:' '^[[:blank:]]\+\([^<]\+\)[[:blank:]]\+<' '3' <<< "$PAGE" | \
+        html_to_utf8) || return
+    readonly FILE_NAME
+
+    FILE_ID=$(parse 'data-slow-id' 'data-slow-id="\([^"]\+\)"' <<< "$PAGE") || return
+
+    # request download
+    PAGE=$(curl -b "$COOKIE_FILE" -d "slow_id=$FILE_ID" \
+        "$URL") || return
+
+    # check for forced delay
+    WAIT=$(parse_quiet 'Please wait .* to download this file' \
+        'wait \([[:digit:]:]\+\) to download' <<< "$PAGE")
+
+    if [ -n "$WAIT" ]; then
+        local HOUR MIN SEC
+
+        HOUR=${WAIT%%:*}
+        SEC=${WAIT##*:}
+        MIN=${WAIT#*:}; MIN=${MIN%:*}
+        log_error 'Forced delay between downloads.'
+        # Note: Get rid of leading zeros so numbers will not be considered octal
+        echo $(( (( ${HOUR#0} * 60 ) + ${MIN#0} ) * 60 + ${SEC#0} ))
+        return $ERR_LINK_TEMP_UNAVAILABLE
+    fi
+
+    # Free user can't download large files.
+    if match "Free user can't download large files" "$PAGE"; then
+        log_error 'Free user cant download large files'
+        return $ERR_LINK_NEED_PERMISSIONS
+    fi
+
+    # check for and handle CAPTCHA (if any)
+    if match 'captcha.html' "$PAGE"; then
+        log_debug 'Captcha found'
+        local CAPTCHA_URL IMG_FILE
+        local RESP WORD ID CAPTCHA_DATA
+
+        # Get captcha image
+        CAPTCHA_URL=$(parse_attr ' id="captcha_' 'src' <<< "$PAGE") || return
+        IMG_FILE=$(create_tempfile '.fboom_me.png') || return
+        curl -b "$COOKIE_FILE" -o "$IMG_FILE" "$BASE_URL$CAPTCHA_URL" || return
+
+        # Solve captcha
+        # Note: Image is a 260x80 png file containing 6-7 characters
+        RESP=$(captcha_process "$IMG_FILE" fboom_me 6 7) || return
+        { read WORD; read ID; } <<< "$RESP"
+        rm -f "$IMG_FILE"
+
+        CAPTCHA_DATA="-d CaptchaForm%5Bcode%5D=$WORD"
+
+        log_debug "Captcha data: $CAPTCHA_DATA"
+
+        FILE_ID=$(parse_form_input_by_name 'uniqueId' <<< "$PAGE") || return
+
+        PAGE=$(curl -b "$COOKIE_FILE" $CAPTCHA_DATA \
+            -d 'free=1' -d 'freeDownloadRequest=1' \
+            -d "uniqueId=$FILE_ID" \
+            -H 'X-Requested-With: XMLHttpRequest' \
+            "$URL") || return
+
+        if match 'The verification code is incorrect' "$PAGE"; then
+            log_error 'Wrong captcha'
+            captcha_nack "$ID"
+            return $ERR_CAPTCHA
+        fi
+
+        log_debug 'Correct captcha'
+        captcha_ack "$ID"
+
+        # parse wait time
+        WAIT=$(parse ' id="download-wait-timer"' \
+            '.*>[[:blank:]]*\([[:digit:]]\+\)' 1 <<< "$PAGE") || return
+
+        if [ -n "$WAIT" ]; then
+            wait $(( WAIT + 1 )) || return
```

Comment:
This if statement will always be true because above you have '|| return'. So it can be removed. Also are you sure that '$(( WAIT + 1 ))' is necessary? I have never been using this in my modules and they work properly. Here I would write simply without if statement:
```
wait $WAIT || return
```